### DISCUSSION TITLE
This Serverless Function has crashed.

### DISCUSSION THREAD

**Discussion by AleksandrSanduliak** [vercel/community#1362] [Help] [answered]
i want to deploy my project, but every time it crashes due to "500: INTERNAL_SERVER_ERROR". 

git : https://github.com/AleksandrSanduliak/crab

error log running checks :
{
  "stack": "Error: expect(received).toBeLessThan(expected)\n\nExpected: < 400\nReceived:   500\n    at BaseHandler.apply (/var/task/src/2022-10/node_modules/vm2/lib/bridge.js:479:11)\n    at /tmp/q9miit8ldd4apka/script.js:17:31",
  "message": "expect(received).toBeLessThan(expected)\n\nExpected: < 400\nReceived:   500"
}

**Comment by AleksandrSanduliak** [comment]
its work fine if i use standart html file, else if using ejs template every time error

**Comment by amyegan** [comment, marked as answer]
This kind of error happens when you try to run a server instead of static content. The Vercel platform is made for static sites and Serverless Functions. 

There is a guide about [using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel) that may help you get Express running in a Serverless Function. Just be sure to review the [Drawbacks and Edge Cases](https://vercel.com/guides/using-express-with-vercel#drawbacks-and-edge-cases) to get a better understanding of the limitations with this approach

**Reply by AbGrS** [reply]
Well that guide doesn't work. After wasting a few hours, I think vercel is useless. Gotta find something better.

**Reply by amyegan** [reply]
I built a minimal example repo based on an older version that guide. Sharing here in case it helps anyone: https://github.com/amyegan/express-vercel

**Reply by AbGrS** [reply]
Here's I tried deploying a sample app using that guide:
https://sample-real-estate-backend-iotupsklz-rahul-dwivedis-projects.vercel.app/

Followed countless other suggestions too to resolve the error but nothing worked. To begin with, why's there no option to even select 'express' (or Node) framework on vercel? You need to select 'other'.

Then found this site 'Render' and in less than 5 minutes I could get my app deployed and working.

**Reply by amyegan** [reply]
I'm not able to replicate what went wrong in your case. I just built a new example from the steps in the guide and deployed it at https://express-1362.vercel.app/ without any errors. 

However, I'm glad you were able to find an alternative solution that worked for you 😄
